### PR TITLE
[OpenVINO Backend] Relocate test exclusions to concrete list

### PR DESCRIPTION
**Pull request by goyaladitya05**
This PR improves test coverage and transparency for the OpenVINO backend by transitioning from broad file-level/directory-level exclusions to granular, method-level exclusions.

Previously, many core test suites were completely skipped in the OpenVINO CI via excluded_tests.txt. This led to a lack of visibility into which specific operations were failing and prevented the verification of passing logic within those files.

By running these tests locally with the OpenVINO backend and identifying specific failures, we have relocated the problematic cases to excluded_concrete_tests.txt. This allows all other passing tests within these suites to run in every CI cycle.

## Changes
- keras/src/backend/openvino/excluded_tests.txt: Removed 28 file/directory level entries.
- keras/src/backend/openvino/excluded_concrete_tests.txt: Added ~500 specific test cases (e.g., specific Trainer methods, bfloat16 weight loading, and complex layer parameterizations) that are currently blocked.
- Also removed 10 tests related to #22292. They should've been excluded in that PR itself, but were missed by me.

### Affected Test Suites (Reintegrated):
- Layers: attention, rnn, convolutional/conv_transpose, core/dense, core/einsum_dense, core/embedding, core/reversible_embedding, normalization/spectral_normalization, reshaping/reshape, up_sampling*, zero_padding*, layer_test.py.
- Infrastructure: models, saving, optimizers, metrics, losses, regularizers, quantizers, utils.
- Training: trainers/trainer_test.py, compile_utils_test.py, epoch_iterator_test.py.
- Random: seed_generator_test.py.

## Impact
- Thousands of additional individual tests are now being verified in the OpenVINO CI.
- The excluded_tests.txt file has been minimized to only 4 entries (activations, preprocessing layers, regularization layers, and legacy), which either hang or have 0% pass rate currently.
- excluded_concrete_tests.txt now serves as a granular roadmap for future OpenVINO backend implementations.

## Verification
Verified locally by running the affected test suites with the OpenVINO backend:
```bash
$env:KERAS_BACKEND="openvino"; .venv\Scripts\python -m pytest keras/src/backend/common/dtypes_test.py keras/src/callbacks/early_stopping_test.py keras/src/dtype_policies/dtype_policy_map_test.py keras/src/layers/convolutional/separable_conv_test.py keras/src/layers/normalization/unit_normalization_test.py keras/src/layers/pooling/average_pooling_test.py keras/src/layers/pooling/max_pooling_test.py

### CODE REVIEW

**Comment by shriramThakare3** [comment]
#22298 
tried fixing this issue 
in original code , negative_order_flag was effectively a mirror of positive_order_flag. Because both checked for a step of exactly 1, any descending list where the step is -1, would trigger both flags to false, failing the continuity check incorrectly

**Comment by goyaladitya05** [comment]
> Thanks! This is great.
> 
> Can you rebase? I tried to do the merge in github but apparently I failed.

Hi @hertschuh ,
I've rebased. Apperently `keras/src/backend/common/dtypes_test.py::DtypesTest::test_empty_lub_in_least_upper_bound` was failing, I've added to it the exclusions for now, to limit the scope for this PR.

**Comment by goyaladitya05** [comment]
Please do not merge this right now, I'm adding more of these test exclusions within an hour or two.

File: keras/src/backend/openvino/excluded_concrete_tests.txt [hunk 1]
```diff
@@ -1,143 +1,687 @@
-
-NumpyDtypeTest::test_angle
```
Comment (goyaladitya05) [inline keras/src/backend/openvino/excluded_concrete_tests.txt:2]: I've re-arranged the file alphabetically, just like it was before.

**Comment by goyaladitya05** [comment]
Hi @hertschuh, this can be merged now. Thanks!
